etcd3/stateful/watch.py

```diff
@@ -240,6 +240,26 @@ def onEvent(self, filter_or_cb, cb=None):
             raise TypeError('callback should be a callable')
         self.callbacks.append((self.get_filter(filter), filter, cb))
 
+    @check_param(at_least_one_of=['filter', 'cb'])
+    def unEvent(self, filter=None, cb=None):  # noqa
+        """
+        remove a callback or filter event that's been previously added via onEvent()
+        If both parameters are given they are ANDd together; to OR the, make two calls.
+
+        :type filter: callable or regex string or EventType
+        :param filter: the callable filter or regex string or EventType the event to be removed was registerd with
+        :param cb: the callback funtion the event to be removed was registerd with
+        """
+        mutable = self.callbacks[:]
```

Comment:
considered that again, if two thread call `unEvent()` the same time, and both thread got switched here, one of the threads will fail to remove the event listener. So, we still need a lock.

Reply:
ohh... good point. Okay, back to a lock it is.